go test ./... fails

`go test ./...` fails for various reasons

One of them is dependency problems.
```
ipfs-1-to-2/go-datastore/lru/datastore.go:6:2: cannot find package "github.com/ipfs/fs-repo-migrations/ipfs-1-to-2/golang-lru" ...
ipfs-1-to-2/goprocess/context/context.go:5:2: cannot find package "golang.org/x/net/context" .. 
ipfs-3-to-4/flatfs/flatfs_test.go:12:2: cannot find package "github.com/ipfs/go-datastore/flatfs" ...
...
```
Note, that one of the ideas (I think) behind fs-repo-migrations was that all dependences should be checked in (@whyrusleeping is this correct?)

Also note that one of the failed dependence is with a test of a dependency.

Another are broken tests:
```
ipfs-4-to-5/go-ds-flatfs/convert_test.go:18: undefined: tempdir
ipfs-4-to-5/go-ds-flatfs/convert_test.go:60: undefined: tempdir
ipfs-4-to-5/go-ds-flatfs/convert_test.go:133: undefined: tempdir
ipfs-4-to-5/go-ds-flatfs/convert_test.go:168: undefined: tempdir
?       github.com/ipfs/fs-repo-migrations/ipfs-4-to-5  [no test files]
ok      github.com/ipfs/fs-repo-migrations/ipfs-4-to-5/go-datastore     0.007s
ok      github.com/ipfs/fs-repo-migrations/ipfs-4-to-5/go-datastore/query       0.003s
FAIL    github.com/ipfs/fs-repo-migrations/ipfs-4-to-5/go-ds-flatfs [build failed]
?       github.com/ipfs/fs-repo-migrations/ipfs-4-to-5/migration        [no test files]
```
Note that the broken test is inside a dependency.

Note the README has this (under Developing Migrations):

> The process should be:
> - Frozen. After the tool is written, all code must be frozen and vendored.

So it may not be kosher to fix old code.

--- a/ipfs-1-to-2/go-migrate/cli.go
+++ b/ipfs-1-to-2/go-migrate/cli.go
@@ -29,10 +29,10 @@ func Run(m Migration) error {
 
 	if !m.Reversible() {
 		if f.Revert {
-			return fmt.Errorf("migration %d is irreversible", m.Versions())
+			return fmt.Errorf("migration %s is irreversible", m.Versions())
 		}
 		if !f.Force {
-			return fmt.Errorf("migration %d is irreversible (use -f to proceed)", m.Versions())
+			return fmt.Errorf("migration %s is irreversible (use -f to proceed)", m.Versions())
 		}
 	}
 

--- a/ipfs-2-to-3/migration/migration.go
+++ b/ipfs-2-to-3/migration/migration.go
@@ -335,7 +335,7 @@ func loadOldKeys(from dstore.Datastore, k dstore.Key) ([]u.Key, error) {
 
 	vb, ok := v.([]byte)
 	if !ok {
-		return nil, fmt.Errorf("pinset %s was not stored as []byte")
+		return nil, fmt.Errorf("pinset %s was not stored as []byte", v)
 	}
 
 	var keys []u.Key

